### PR TITLE
:sparkles: Use Kafka KRaft

### PR DESCRIPTION
**Pull request by clyang82**
## Summary

Use KRaft mode since it is GAed. besides it, also address the following issues in this PR:
1. remove plain listener for Kafka to disable anonymous access
2. 

## Related issue(s)

Fixes # https://issues.redhat.com/browse/ACM-15625

## Tests
* [ ] Unit/function tests have been added and incorporated into `make unit-tests`.
* [ ] Integration tests have been added and incorporated into `make integration-test`.
* [ ] E2E tests have been added and incorporated into `make e2e-test-all`.
* [ ] List other manual tests you have done.

### CODE REVIEW

File: samples/config/confluent_config.go [hunk 1]
```diff
@@ -219,8 +219,8 @@ func GetConfluentConfigMapByUser(c client.Client, namespace, clusterName, userNa
 	cm := config.GetBasicConfigMap()
 	for _, condition := range kafkaCluster.Status.Conditions {
 		if *condition.Type == "Ready" && *condition.Status == "True" {
-			clusterCaCert := kafkaCluster.Status.Listeners[1].Certificates[0]
-			_ = cm.SetKey("bootstrap.servers", *kafkaCluster.Status.Listeners[1].BootstrapServers)
+			clusterCaCert := kafkaCluster.Status.Listeners[0].Certificates[0]
```
Comment (yanmxa) [inline samples/config/confluent_config.go:222]: There are two listeners for the kafka connection. the first one is internal, and the other is external. You mean only keep one type?

**Comment by ldpliu** [comment]
Did you test if the upgrade works well?

File: samples/config/confluent_config.go (hunk 1 above)
Comment (clyang82) [reply samples/config/confluent_config.go:222]: Yes. remove plain listener for Kafka to disable anonymous access.

**Comment by clyang82** [comment]
> Did you test if the upgrade works well?

will handle the upgrade in a separate user story.

**Comment by yanmxa** [comment]
It's a significant improvement in transport optimization, especially in its footprint!

/lgtm

**Pull request merged**